src/Compilers/CSharp/Test/Emit/Emit/InAttributeModifierTests.cs

```diff
@@ -4413,10 +4413,53 @@ public class Test
     public ref readonly int M() => throw null;
 }";
 
-            CreateCompilation(user, references: new[] { ref1, ref2 }).VerifyDiagnostics(
-                // (4,12): error CS0518: Predefined type 'System.Runtime.InteropServices.InAttribute' is not defined or imported
-                //     public ref readonly int M() => throw null;
-                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "ref readonly int").WithArguments("System.Runtime.InteropServices.InAttribute").WithLocation(4, 12));
+            var comp = CreateCompilation(user, references: new[] { ref1, ref2 }).VerifyDiagnostics();
+
+            // we prefer the type from corlib
+            var m = comp.GlobalNamespace.GetTypeMember("Test").GetMethod("M");
+            var attribute = m.RefCustomModifiers.Single().Modifier;
+            Assert.NotNull(attribute.ContainingAssembly.GetTypeByMetadataName("System.Object"));
```

Comment:
>Assert.NotNull [](start = 12, length = 14)

What are we verifying? Won't `GetTypeByMetadataName()` look though all references to find "System.Object"?

Reply:
This is verifying that the attribute we found came from corlib.

---
In reply to: [586780243](https://github.com/dotnet/roslyn/pull/51571#discussion_r586780243) [](ancestors = 586780243)